lib/src/main/kotlin/com/swmansion/starknet/data/types/MerkleTree.kt

```diff
@@ -12,9 +13,9 @@ data class MerkleTree(
 
     companion object {
         @JvmStatic
-        fun hash(a: Felt, b: Felt): Felt {
+        fun hash(a: Felt, b: Felt, hashFunction: HashMethod): Felt {
             val (aSorted, bSorted) = if (a < b) Pair(a, b) else Pair(b, a)
-            return StarknetCurve.pedersen(aSorted, bSorted)
+            return hashFunction.hash(aSorted, bSorted)
```

Comment:
hmm, I'm leaning towards ditching the defaults altogether tbh. I'm worried the defaults could cause us unnecessary trouble, by ditching we give all the power (and responsibility) to users. They will have to check which hash they need, but at least this will be an informed choice they will de before using the function (and not after).